internal/representation/list.go

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package representation
+
+import (
+	"bytes"
+	"compress/flate"
+	"compress/gzip"
+	"compress/zlib"
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"io"
+	"net/url"
+
+	"github.com/freerware/negotiator/internal/header"
+	rep "github.com/freerware/negotiator/representation"
+	"gopkg.in/yaml.v2"
+)
+
+type representation struct {
```

Comment:
add comment for `representation`, as well as for all other types and methods in this file.

Reply:
done!